app/controllers/admin/authors_controller.rb

```diff
@@ -0,0 +1,62 @@
+class Admin::AuthorsController < ApplicationController
+  include Authorizable
+
+  layout "admin_application"
+
+  before_action :authorize_librarian
+  before_action :check_author, except: %i(index new create)
+
+  def index
+    @pagy, @authors = pagy Author.all
+  end
+
+  def new
+    @author = Author.new
+  end
+
+  def show; end
+
+  def create
+    @author = Author.new author_params
+    if @author.save
+      redirect_to admin_author_url(@author)
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @author.update author_params
+      flash[:success] = t ".updated"
+      redirect_to [:admin, @author]
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    if @author&.destroy
```

Comment:
chỗ này cần destroy luôn hỉ, vì em có dùng before filter để kiểm tra cái thằng author tồn tại rồi á
```suggestion
    if @author.destroy
```